### PR TITLE
Add rootless running

### LINKED ISSUE #34: Implement rootless running
Allow non-root users to run tests which don't require privileges.
This is already covered with the test collection, by tracking tests which use privileged file types or annotated with root requirement. We just need to implement this on the runner.

pjd/pjdfstest#65

### PR DESCRIPTION
**Pull request by musikid**
Skip tests requiring root privileges.

Close #34

### CODE REVIEW

File: rust/src/main.rs [hunk 1]
```diff
@@ -133,9 +135,11 @@ fn main() -> anyhow::Result<()> {
 }
 
 /// Run provided test cases and filter according to features and flags availability.
+//TODO: Refactor this function
 fn run_test_cases(
     test_cases: &Vec<&TestCase>,
     verbose: bool,
+    is_root: bool,
```
Comment (asomers) [inline rust/src/main.rs:142]: Why take this as an argument, rather than just looking up the UID from within this function?

**Comment by asomers** [comment]
Also, it would be great to test this feature in CI.  You can do that by adding a step that runs `su -m nobody -c "target/release/pjdfstest -c $CFG"`.

File: rust/src/main.rs (hunk 1 above)
Comment (musikid) [reply rust/src/main.rs]: This was just a mistake.

File: .cirrus.yml [hunk 2]
```diff
@@ -9,6 +9,8 @@ build: &BUILD
     - cargo test
     - cargo build --all-targets --release
     - target/release/pjdfstest -c $CFG
+    - mkdir nobody && chown nobody nobody
+    - su -m nobody -c "target/release/pjdfstest -c $CFG -p nobody"
```
Comment (asomers) [inline .cirrus.yml:13]:
```suggestion
    - target/release/pjdfstest -c $CFG -p /tmp
    - su -m nobody -c "target/release/pjdfstest -c $CFG -p /tmp"
```

**Review by asomers — CHANGES_REQUESTED** [review]

File: .cirrus.yml [hunk 3]
```diff
@@ -8,9 +8,8 @@ build: &BUILD
     - cd rust
     - cargo test
     - cargo build --all-targets --release
-    - target/release/pjdfstest -c $CFG
-    - mkdir nobody && chown nobody nobody
-    - su -m nobody -c "target/release/pjdfstest -c $CFG -p nobody"
+    - target/release/pjdfstest -c $CFG -p /tmp
+    - su -m nobody -c "target/release/pjdfstest -c $CFG -p /tmp"
```
Comment (musikid) [inline .cirrus.yml:12]: I'm not sure on why it doesn't work. I'm able to successfully run it on my machine, but it fails on every Docker image that I tested.

Comment (musikid) [reply .cirrus.yml:12]: Yes, it is available.

Comment (asomers) [reply .cirrus.yml:12]: I don't know why you're getting this error.  The `-m` should allow it.  But what if you try adding a `-s /bin/sh` option to `su`?

**Review by asomers — APPROVED** [review]

File: .cirrus.yml (hunk 3 above)
Comment (musikid) [reply .cirrus.yml]: It works now. I used an environment variable for su args since `-s` is unsupported on FreeBSD. A little ugly, but it works!

**Pull request merged**